### PR TITLE
Put combine update in main thread to fix gif playback freezing issue when open too many gif

### PR DESCRIPTION
**Pull request by hm1113**
After putting some combine related update in main thread, I no longer see the crash or app freezing

### CODE REVIEW

File: SDWebImageSwiftUI/Classes/ImagePlayer.swift [hunk 1]
```diff
@@ -87,12 +87,16 @@ public final class ImagePlayer : ObservableObject {
...
-                self?.currentFrameIndex = index
-                self?.currentFrame = frame
-            }
-            imagePlayer.animationLoopHandler = { [weak self] (loopCount) in
-                self?.currentLoopCount = loopCount
+                // Make sure we trigger combine in the main thread
+                DispatchQueue.main.async {
+                    self?.currentFrameIndex = index
+                    self?.currentFrame = frame
+                }
```
Comment (renup) [inline SDWebImageSwiftUI/Classes/ImagePlayer.swift:94]: down below line 107, seems like we are assigning the runloop mode.. by default it is .common, may be change it there to .main and you might not have to make a dispatchQueue.main block here?

BTW, why are we putting things in main run loop?

Comment (hm1113) [reply SDWebImageSwiftUI/Classes/ImagePlayer.swift:94]: If you look at SDWebImageSwiftUI comment, you'll see this line:
```
This is useful for some cases, for example, always specify NSDefaultRunLoopMode, if you want to pause the animation when user scroll (for Mac user, drag the mouse or touchpad)
```

we use default to to pause animation when scrolling.

currentFrame is a published object, looks like some times it would randomly freeze if we don't update in UI thread. After making this change, I don't see the issue any more.

File: SDWebImageSwiftUI/Classes/ImagePlayer.swift [hunk 2]
```diff
@@ -38,7 +38,7 @@ public final class ImagePlayer : ObservableObject {
     @Published public var currentFrame: PlatformImage?
     
     /// Current playing frame index
-    @Published public var currentFrameIndex: UInt = 0
+    public var currentFrameIndex: UInt = 0
```
Comment (anton-rodzik-os) [inline SDWebImageSwiftUI/Classes/ImagePlayer.swift:41]: What is the purpose of this change?

Comment (hm1113) [reply SDWebImageSwiftUI/Classes/ImagePlayer.swift:41]: I see in this callback animationFrameHandler, it updates two published fields and each time will trigger two passes of UI re-rendering, it seems to be inefficient. So I change it to only update once for each gif frame update:

Looks like we only need this currentFrame to render animation.

It's more like some performance optimization.

**Review by renup — APPROVED** [review]
